### PR TITLE
[webui] React setup script

### PR DESCRIPTION
**Pull request by saganshul**
React setup instructions added to installation script.

## Reviewers
@delta24 @anantshri @tunnelshade @7a 

## How Has This Been Tested?
This is tested on ubuntu.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### CODE REVIEW

File: install/install.py [hunk 1]
```diff
@@ -110,6 +111,40 @@ def is_compatible():
             return True
 
 
+def setup_react_stack():
+    """Install the node dependencies listed in package.json and build the bundle file using webpack.
+
+    """
```
Comment (delta24) [inline install/install.py]: Docstring ends on the same line for a single-line comment.

**Comment by delta24** [comment]
You should add this to the `distro_independent.cfg` file and it will called by the install script. Add an install script `setup_react.sh` which does the same thing as `setup_react_stack` and call it in the `distro_independent.cfg` like:

```
[React setup]
directory = /tmp/owtf-install/react_setup/%(Pid)s
command = sh %(RootDir)s/install/setup_react.sh
```

File: install/setup_react.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env sh
+#
+# This script install the node dependencies listed in package.json and build the bundle file using webpack.
+
+# bring in the color variables: `normal`, `info`, `warning`, `danger`, `reset`
+. "$(dirname "$(readlink -f "$0")")/utils.sh"
+
+sudo apt-get install curl
```
Comment (delta24) [inline install/setup_react.sh]: This should not be there.

Comment (saganshul) [inline install/setup_react.sh]: So can we assume that curl is installed?

Comment (delta24) [inline install/setup_react.sh]: Yes + should not assume that it is a Debian-based distro.

**Comment by saganshul** [comment]
so you just want me to remove this line right?

File: install/setup_react.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# This script install the node dependencies listed in package.json and build the bundle file using webpack.
+
+# bring in the color variables: `normal`, `info`, `warning`, `danger`, `reset`
+. "$(dirname "$(readlink -f "$0")")/utils.sh"
+
+# Instead of using apt-get to install npm we will nvm to install npm because apt-get installs older-version of node
+echo "${normal}[*] Installing npm using nvm.${reset}"
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.1/install.sh | bash
+nvm install node
+nvm alias default node
```
Comment (delta24) [inline install/setup_react.sh]: Don't use pipe to use on a script. This is a bad practice for an infosec tool :D

**Comment by delta24** [comment]
Oops! my mistake :(
Change the react_setup directory in distro_independent.cfg to be `/tmp/owtf-install/react_setup` --> This is so that the installer installs it by default.

**Pull request merged**